### ISSUE TITLE
dbt model names are not always unique

### ISSUE THREAD

**Issue by owlas** [lightdash/lightdash#8066]
### Description

Lightdash assumes that dbt model names are unique. This was true until this PR was merged: https://github.com/dbt-labs/dbt-core/pull/7374 - with this change, Project A can have model `users` and **also** have a dependency on Project B with a model called `users`

The resulting manifest will have two models with the name `users` but they'll have different unique ids `model.project_a.users` and `model.project_b.users`

**Comment by TuringLovesDeathMetal** [comment]
There is also versioning models in dbt which causes this issue! https://docs.getdbt.com/docs/collaborate/govern/model-versions#how-to-create-a-new-version-of-a-model

**Comment by TuringLovesDeathMetal** [comment]
We should potentially make the unique ID a different thing than the name (before, teh name used to be unique, but now it's not anymore - we should update the model's unique ID to be something other than the name. E.g. project.model_name)

**Comment by ZeRego** [comment]
In this [PR](https://github.com/lightdash/lightdash/pull/8390) I'm avoiding this error by discarding/ignoring explores with duplicate names. It will only use the first explore for each name.

This fix is focused on not throwing an error to unblock users that deploy their project via the UI which deploys all their models. We can open a new ticket to support multiple explores with the same name.
